crates/meroctl/src/cli/context/create.rs

```diff
@@ -0,0 +1,179 @@
+use calimero_primitives::application::ApplicationId;
+use calimero_primitives::identity::Context;
+use calimero_server_primitives::admin::ApplicationListResult;
+use camino::Utf8PathBuf;
+use clap::Parser;
+use eyre::eyre;
+use reqwest::{Client, Url};
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use tracing::info;
+
+use super::get_ip;
+use crate::cli::RootArgs;
+use crate::config::ConfigFile;
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateContextRequest {
+    application_id: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CreateContextResponse {
+    data: Context,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InstallApplicationRequest {
+    pub application: ApplicationId,
+    pub version: Version,
+    pub path: Option<Utf8PathBuf>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ListApplicationsResponse {
+    data: ApplicationListResult,
+}
+
+#[derive(Debug, Parser)]
+pub struct CreateCommand {
+    /// The application ID to attach to the context
+    #[clap(long, short = 'a')]
+    application_id: Option<String>,
+
+    /// Enable dev mode
+    #[clap(long)]
+    dev: bool,
+
+    /// Path to use in dev mode (required in dev mode)
+    #[clap(short, long, requires = "dev")]
+    path: Option<Utf8PathBuf>,
+
+    /// Version of the application (required in dev mode)
+    #[clap(short, long, requires = "dev")]
+    version: Option<Version>,
+}
+impl CreateCommand {
+    pub async fn run(self, root_args: RootArgs) -> eyre::Result<()> {
+        let path = root_args.home.join(&root_args.node_name);
+        if ConfigFile::exists(&path) {
+            if let Ok(config) = ConfigFile::load(&path) {
+                let multiaddr = config
+                    .network
+                    .server
+                    .listen
+                    .first()
+                    .ok_or_else(|| eyre!("No address."))?;
+                let base_url = get_ip(multiaddr)?;
+
+                match self.dev {
+                    true => Ok(link_local_app(base_url, self.path, self.version).await?),
+                    false => Ok(create_context(base_url, self.application_id).await?),
+                }
+            } else {
+                Err(eyre!("Failed to load config file"))
+            }
+        } else {
+            Err(eyre!("Config file does not exist"))
+        }
+    }
+}
+
+async fn create_context(base_url: Url, application_id: Option<String>) -> eyre::Result<()> {
+    let application_id =
+        application_id.ok_or_else(|| eyre!("Application ID is required for starting context"))?;
```

Comment:
this should be encoded in clap, so the requirements validation is unified by clap

explore `Arg` options on how to specify inter-argument relationships & dependence

https://docs.rs/clap/latest/clap/struct.Arg.html